debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerChangeTablePointer.java

```diff
@@ -47,16 +48,26 @@
     private ResultSetMapper<Object[]> resultSetMapper;
     private final ResultSet resultSet;
     private final int columnDataOffset;
+    private final SourceTimestampMode sourceTimestampMode;
 
-    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet) {
-        this(changeTable, resultSet, COL_DATA);
+    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, SourceTimestampMode sourceTimestampMode) {
+        this(changeTable, resultSet, sourceTimestampMode, COL_DATA);
     }
 
-    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, int columnDataOffset) {
+    public SqlServerChangeTablePointer(SqlServerChangeTable changeTable, ResultSet resultSet, SourceTimestampMode sourceTimestampMode, int columnDataOffset) {
         super(changeTable, resultSet, columnDataOffset);
         // Store references to these because we can't get them from our superclass
         this.resultSet = resultSet;
         this.columnDataOffset = columnDataOffset;
+        this.sourceTimestampMode = sourceTimestampMode;
+    }
+
+    public Timestamp getCommitTimestamp() throws SQLException {
+        if (SourceTimestampMode.PROCESSING.equals(sourceTimestampMode)) {
+            return null;
```

Comment:
As far as I understand, this code is never meant to be executed. We could think of a cleaner API. Like the following instance:
```
function getTimestamp(mode, resultSet, clock) {
  if (mode == /*... */) {
    return /* ... */
  }

  return /* ... */
}
```
This API could be even made part of the `SourceTimestampMode` interface (i.e. each mode would implement its own logic of determining the timestamp).

Reply:
I'm confused. What is the point of passing `mode` as an argument of `getTimestamp` if it's already available as the object's property? Same for `resultSet`. Also there is no `clock.getTimestamp()` method, there is `clock.currentTime()` which returns `java.time.Instant`.

`SourceTimestampMode` is an enum. Do you want to refactor it into an interface, two classes and a factory method? IMO it's going to blur the area of responsibility of `SourceTimestampMode` which is configuration sourcing.

Reply:
> I'm confused. What is the point of passing `mode` as an argument of `getTimestamp` if it's already available as the object's property?

It's a pseudo-code. The primary point is to not have the code that is not meant to be executed, returning a NULL value that nobody will expect.

> `SourceTimestampMode` is an enum. Do you want to refactor it into an interface, two classes and a factory method?

Enums in Java are more of objects than scalar values. See `BinaryHandlingMode` for example.

> IMO it's going to blur the area of responsibility of SourceTimestampMode which is configuration sourcing.

Since those are objects, their responsibilities are pretty much clear:
1. Map a given mode (the value of the enum) to the string from the configuration.
2. Implement the mode-specific logic which is:
   1. Build the SQL that provides all the data in the given mode.
   2. Provide the timestamp corresponding to the given change row.

This approach is more robust than if-statements scattered across the codebase:
1. Each method of each mode can be unit-tested (isolated from the rest of the code).
2. When adding a new method, one will not forget to add another if-statement somewhere because all the mode-related code is in one place.
3. The connector itself doesn't contain any mode-specific logic, it's just an abstract interface.

Reply:
I've moved this logic to `SourceTimestampMode`.